index.js

```diff
@@ -12,13 +12,85 @@
 //so in general the game is a loop that takes a player turn and asks for the cordinates for the oponent's ships. loop is over when one of the players has 0 ships.
 //return value contains a string with the winner name
 
+const player1 = {
+  name: prompt(`What is player1's name?`),
+  ships: 0,
+  gameBoard: [[0,0,0,0],[0,0,0,0],[0,0,0,0],[0,0,0,0]]
+};
+const player2 = {
+  name: prompt(`What is player2's name?`),
+  ships: 0,
+  gameBoard: [[0,0,0,0],[0,0,0,0],[0,0,0,0],[0,0,0,0]]
+};
+
+
+let player1Turn = true;
 
 battleship = () => {
-  return
+
+  // for(let i = 0; i < 4; i ++){
+  //   Math.floor(Math.random() * Math.floor(5));
+  // }
+
+  while(player1.ships !== 4){
+    const qX = Math.floor(Math.random() * Math.floor(5));
+    const qY = Math.floor(Math.random() * Math.floor(5));
+    if( player1.gameBoard[qY][qX] == 0){
+      player1.gameBoard[qY][qX] = 1;
+      player1.ships++
+    }
+  }
+  while( player2.ships !== 4 ){
+    const qX = Math.floor(Math.random() * Math.floor(5));
+    const qY = Math.floor(Math.random() * Math.floor(5));
+    if( player2.gameBoard[qY][qX] == 0){
+      player2.gameBoard[qY][qX] = 1;
+      player2.ships++
+    }
+  }
+  
+  while(player1.ships !== 0 && player2.ships !== 0){
+    const curentPlayer = player1Turn ? player1.name : player2.name;
+
+    const yL = prompt(`${curentPlayer} pick a location to strike! First enter the Y location 0-3
+          Y
+    X         0   1   2   3
+          0 
+          1  
+          2   
+          3   `);
```

Comment:
Just my personal feedback, but this confused me at first because I was comparing the array structure to your diagram structure and the rows/columns are actually opposite:

Array:
```javascript
[
  [0, 0, 1, 0],
  [1, 0, 0, 0],
  [0, 0, 0, 0],
  [0, 1, 0, 1]
]
```

Diagram:
```javascript
          Y
    X         0   1   2   3
          0 
          1  
          2   
          3
```

So I was providing this input: `y = 2` and `x = 0`, thinking that I was going to hit the ship on the first row - but instead it was a miss.

If you look at the two-dimensional array like it's a table, then `x` represents the columns and `y` represents the rows (this is opposite from your diagram).

You could either switch the coordinate labels or flip the array indices:
`player2.gameBoard[qY][qX]` -> `player2.gameBoard[qX][qY]`

Reply:
fixed